src/hiro_graph_client/clientlib.py

```diff
@@ -380,22 +261,69 @@ def _get_query_part(params: dict) -> str:
         params_cleaned = {k: v for k, v in params.items() if v is not None}
         return ('?' + urlencode(params_cleaned, quote_via=quote, safe="/,")) if params_cleaned else ""
 
-    def _parse_json_response(self, res: requests.Response, token: str = None) -> dict:
+    def _parse_json_response(self, res: requests.Response) -> dict:
         """
         Parse the response of the backend.
 
         :param res: The result payload
-        :param token: The external token if provided
         :return: The result payload
         :raises RequestException: On HTTP errors.
         """
         try:
-            self._check_response(res, token)
+            self._check_response(res)
             self._check_status_error(res)
             return res.json()
         except (json.JSONDecodeError, ValueError):
             return {"error": {"message": res.text, "code": 999}}
 
+    def _log_communication(self, res: requests.Response, request_body: bool = True, response_body: bool = True) -> None:
+        """
+        Log communication that flows across requests' methods. Contains options to disable logging of the body portions
+        of the requests to avoid dumping large amounts of data and breaking streaming of results.
+
+        This log does not log an exact binary representation of the transmitted bodies but uses the encoding defined
+        in *res* to print it as strings.
+
+        Authorization and cookie headers will be obscured by only displaying the last six characters of their values.
+
+        :param res: The response of a request. Also contains the request.
+        :param request_body: Option to disable the logging of the request_body.
+        :param response_body: Option to disable the logging of the response_body.
+        :return:
+        """
+
+        def _log_headers(headers) -> str:
+            result: str = ""
+            for k, v in headers.items():
+                if k == "Authorization" or k.find("Cookie") != -1:
+                    v = f"(shortened) ...{v[-6:]}"
+                result += f"{k}: {v}\n"
+            return result
+
+        def _body_str(body: Union[str, bytes], encoding: str) -> str:
+            if body is None:
+                return ""
+            if isinstance(body, bytes):
+                body = str(body, encoding)
+            return body
+
+        if res.status_code >= 400 or logging.root.isEnabledFor(logging.DEBUG):
+            log_message = f'''
+---------------- request ----------------
+{res.request.method} {res.request.url}
+{_log_headers(res.request.headers)}
+{_body_str(res.request.body, res.encoding) if request_body else "(body hidden)"}
+---------------- response ----------------
+{res.status_code} {res.reason} {res.url}
+{_log_headers(res.headers)}
+{_body_str(res.text, res.encoding) if response_body else "(body hidden)"}
+'''
```

Comment:
consider using https://github.com/JetBrains/http-request-in-editor-spec/blob/master/spec.md as message format

Reply:
I think I am matching this format already as far as I can see.